modules/dev-server/main.tf

```diff
@@ -42,36 +93,44 @@ resource "aws_ecs_task_definition" "this" {
   volume {
     name = "consul-data"
   }
-  container_definitions = jsonencode([
-    {
-      name      = "consul-server"
-      image     = var.consul_image
-      essential = true
-      portMappings = [
-        {
-          containerPort = 8301
-        },
-        {
-          containerPort = 8300
-        },
-        {
-          containerPort = 8500
-        }
-      ]
-      logConfiguration = var.log_configuration
-      entryPoint       = ["/bin/sh", "-ec"]
-      command          = [replace(local.consul_server_command, "\r", "")]
-      mountPoints = [
-        {
-          sourceVolume  = "consul-data"
-          containerPath = "/consul"
+  container_definitions = jsonencode(concat(
+    local.tls_init_containers,
+    [
+      {
+        name      = "consul-server"
+        image     = var.consul_image
+        essential = true
+        portMappings = [
+          {
+            containerPort = 8301
+          },
+          {
+            containerPort = 8300
+          },
+          {
+            containerPort = 8500
+          }
+        ]
+        logConfiguration = var.log_configuration
+        entryPoint       = ["/bin/sh", "-ec"]
+        command          = [replace(local.consul_server_command, "\r", "")]
+        mountPoints = [
+          {
+            sourceVolume  = "consul-data"
+            containerPath = "/consul"
+          }
+        ]
+        linuxParameters = {
+          initProcessEnabled = true
         }
-      ]
-      linuxParameters = {
-        initProcessEnabled = true
+        dependsOn = var.tls ? [
```

Comment:
should we update containerPorts on line 111

Reply:
I need to keep that one because the load balancer needs it in either tls or no-tls case. We could probably make it conditional on whether lb is enabled, but since it's the dev server I'm not sure if it matters. wdyt?

Reply:
yeah leave it but I was thinking we'd add port 8501 to it

Reply:
We don't really need it though, nothing is connecting to it over that host port.

Reply:
I'm going to resolve this but let me know if you think we need that host IP, and I can add it in a separate PR.

Reply:
Forgot that the clients hit it over RPC, not the HTTPS endpoint. Resolved!